src/engine.c

```diff
@@ -10664,6 +10710,41 @@ void produce_chosen(game *g, int who, int c_idx, int o_idx)
 		return;
 	}
 
+	/* Check for draw per Xeno military world */
+	if (o_ptr->code & P5_DRAW_XENO_MILITARY)
+	{
+		/* Assume no Xeno military worlds */
```

Comment:
Only worlds have the MILITARY flag set, so the code from here to draw_cards is equivalent to
count = count_active_flags(g, who, FLAG_MILITARY | FLAG_XENO);